third-party/cmake/BuildLuv.cmake

```diff
@@ -89,6 +89,10 @@ endif()
 set(LUV_BUILD_COMMAND ${CMAKE_COMMAND} --build .)
 set(LUV_INSTALL_COMMAND ${CMAKE_COMMAND} --build . --target install)
 
+if(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+  set(CMAKE_MAKE_PROGRAM "/usr/local/bin/gmake" CACHE INTERNAL "make->gmake" FORCE)
+endif()
```

Comment:
This seems a bit heavy-handed. In any case, should be done in luv's own CMakeLists.txt, no? It usually only takes 2-3 days to get something merged over there, and we can bump to that version.

Reply:
You're right in pointing out that the while building luv, from the official luv [repository](https://github.com/luvit/luv), you CMAKE_MAKE_PROGRAM is indeed set to /usr/bin/make, but it isn't invoked at any step, whereas here, it is, both for luv and msgpack-c. luv gets build first, so I set it there. Let me know if you'd like me to put it somewhere else.

Reply:
I didn't realize we were using this directly, I see that now. 

> CMAKE_MAKE_PROGRAM is indeed set to /usr/bin/make

Is that perhaps the root problem? 

Nvim requires `gmake`. We even warn about this in [BSDmakefile](https://github.com/neovim/neovim/blob/master/BSDmakefile).

I wonder why `make` is being used at all, on OpenBSD.

If anything, I would guess we should set `CMAKE_MAKE_PROGRAM=gmake` in the top-level CMakeLists.txt. But I wonder why CMake didn't do that for us, or how could it possibly work on other BSDs?

Reply:
>Is that perhaps the root problem?

Yes! The fact that it is not just being set up by cmake, but also being used at an intermediary step. While building luv from it's repository, it just sets it up with the wrong value but it stays unused, so it isn't a problem there.

>Nvim requires gmake. We even warn about this in BSDmakefile.

I have `gmake` on my system, but it still sets it up with the wrong value. However, `make` is available in `/usr/bin` and `gmake` in `/usr/local/bin`. All the packages installed using the package manager are installed to `/usr/local/bin` by default unless you specify a prefix.

>If anything, I would guess we should set CMAKE_MAKE_PROGRAM=gmake in the top-level CMakeLists.txt. 

OK!

>But I wonder why CMake didn't do that for us, or how could it possibly work on other BSDs?

This problem also occurs on DragonflyBSD, I'll push a fix for that. (currently, there are other problems too.)

Reply:
>make is available in /usr/bin and gmake in /usr/local/bin

It might be worth noting that on NetBSD, it `make` was present in `/usr/bin` and `gmake` in `/usr/pkg/bin`, but that still wasn't a problem there and `/usr/bin` always comes before the other two paths in `PATH`.

Reply:
Actually, on OpenBSD, `cmake` doesn't even search for `make` or `gmake`. It just sets up the variable, even if `make` is not present.

Reply:
>If anything, I would guess we should set CMAKE_MAKE_PROGRAM=gmake in the top-level CMakeLists.txt.

Nope, this doesn't stop `CMAKE_MAKE_PROGRAM` from being set to `/usr/bin/make`. In fact, everything, except what I've done is failing. Even setting it to `gmake` in the same file, as current one, isn't working. We need to tell it the full path and exactly where I've done it.

Reply:
> Nope, this doesn't stop CMAKE_MAKE_PROGRAM from being set to /usr/bin/make

Did you `rm -rf build` before trying this? Else it would be cached, this is part of the "generator" step.
https://cmake.org/cmake/help/v3.0/variable/CMAKE_MAKE_PROGRAM.html

Reply:
>Did you rm -rf build before trying this?

Yes! Just to eliminate the ill effects of guessing and to be completely sure.

>Else it would be cached, this is part of the "generator" step.

If you look at the build log, you’ll notice that something is triggering regeneration of cache right before building luv. That why I `FORCE`d the rewrite of `CACHE`.

Reply:
At least `MAKE_PRG` should be correctly set to `gmake` here. So can't that be used instead of hardcoding the path? And it should be done for all BSDs, not just openbsd.

    set(CMAKE_MAKE_PROGRAM ${MAKE_PRG} ...

Reply:
`MAKE_PRG` is not set to `gmake`. It's set to `$(MAKE)`. I believe [this](https://github.com/neovim/neovim/blob/d49c1dda8bf57587c609f3de7b78393f748a9df4/third-party/CMakeLists.txt#L81) is the culprit. If you don't want to hardcode it, we can either search for it here itself, or remove the line that I linked (Although, it says that it's a fix for an error).
For now, I've changed the code a bit. This should work better. The previous version wasn't working when I tested it today.

Reply:
>And it should be done for all BSDs, not just openbsd.
Paths on different BSDs are different. If we were to do it, we'll have to search for it. Additionally, I haven't encountered this error on NetBSD.

Reply:
On a side note, how do you plan on setting up CI for OpenBSD?

Reply:
I’ve removed the hardcoding.